Code/Engine/InputManager.cpp

```diff
@@ -65,45 +115,22 @@ bool sad::InputManager::GetButton(SDL_GameControllerButton button)
     return SDL_GameControllerGetButton(controller, button);
 }
 
-/**
- * @brief Returns true if a specified controller button is pressed in the current frame.
- * @param button
- * @return
-*/
 bool sad::InputManager::GetButtonPressed(SDL_GameControllerButton button)
 {
     if (controller == nullptr)
         return false;
 
-    if (!GetControllerButtonState(button) && SDL_GameControllerGetButton(controller, button)) {
-        UpdateControllerButtonState(button, true);
-        return true;
-    }
-    return false;
+    return buttonState[button] && (buttonUpdateFrames[button] == UpdateCounter);
 }
 
-/**
- * @brief Returns true if a specified controller button is released in the current frame.
- * @param button
- * @return
-*/
 bool sad::InputManager::GetButtonReleased(SDL_GameControllerButton button)
 {
     if (controller == nullptr)
         return false;
 
-    if (GetControllerButtonState(button) && !SDL_GameControllerGetButton(controller, button)) {
-        UpdateControllerButtonState(button, false);
-        return true;
-    }
-    return false;
+    return !buttonState[button] && (buttonUpdateFrames[button] == UpdateCounter);
 }
 
-/**
- * @brief Returns the value rounded to 1.0 of a given axis
- * @param axis 
- * @return 
-*/
 float sad::InputManager::GetAxis(SDL_GameControllerAxis axis)
```

Comment:
This is also a stretch conversation that I left out from the first PR, but it's similar to my comment on adding `KeyCode`.

I believe you can get a controller axis in Unity by doing
```
Input.GetAxis("Horizontal")
or
Input.GetAxis("Vertical")
```

Refactoring `GetAxis()` to take a string and just check the correct `SDL_GameControllerAxis` based on the string might be a simple improvement and would reap similar benefits to what I mention in `KeyCode` conversation.

example? 
```
float sad::InputManager::GetLeftAxis(const std::string& orientation)
{
    SDL_GameControllerAxis axis;

    switch (orientation)
    {
    case "Horizontal":
          axis = SDL_CONTROLLER_AXIS_LEFTX;  // Or whatever the correct SDL enum is
          break;
    case "Vertical":
          axis = SDL_CONTROLLER_AXIS_LEFTY; 
          break;
    default:
          return 0.0f;
    }

    // Handle the axis calcuation using your GetAxis() method that you've already implemented
   return GetAxis(axis);
}
```

Reply:
Added this in for Left and Right joysticks but kept GetAxis public in case a different axis needed to be accessed